### PR TITLE
Revert VideoConfiguration.framerate to a double. Fixes #96

### LINKED ISSUE #12: Do we need to accurately represent non-integer frame rates ?
Non-integer frame rates, such as 24000/1001, are common and aren't precisely representable in binary floating point. Typically, video frame rates are specified as rational numbers or with an enumeration (or both).

We should discuss whether we need to accurately represent the non-integer rates for this API.

### LINKED ISSUE #96: Revert the "DOMString framerate" change from issue #12
The issue of accurately representing "rational integer" time values in JavaScript is much larger than frame rate, and we should push for a general solution.  Above and beyond this general objection, we shouldn't be adding any complicated string parsing logic to represent media times.  The current specification allows the following values as valid `framerate`s:

"1"
"1.5"
"3/2"

But it also considers the following as valid `framerate`s:

"1.5/1"
"3/0.5"

And given that the justification for doing this parsing was to ensure that values could be accurately compared to one another without floating-point rounding, the following (pseudocode) comparisons will all evaluate to false:

```
framerate(1) != framerate("1/1")
framerate(1.5) != framerate("2/1")
framerate("1.") != framerate("1")
framerate("1/1") != framerate("2/2")
```

So we should either completely revert the change, and return the following entry to the `VideoConfiguration` dictionary:

```
    required double framerate;
```

...until such a time where we have an actual "media time" value to be used in place of double.  Or, if we must provide a way to pass a rational value into this API (and I don't believe we do), we should revert the change as above and add an additional property:

```
    double framerateDenominator = 1;
```

### PR DESCRIPTION
**Pull request by mounirlamouri**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/media-capabilities/pull/128.html" title="Last updated on Aug 15, 2019, 12:15 AM UTC (96c22d3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/media-capabilities/128/a6370dc...96c22d3.html" title="Last updated on Aug 15, 2019, 12:15 AM UTC (96c22d3)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -326,23 +326,8 @@ spec: workers; urlPrefix: https://www.w3.org/TR/workers/#
...
-              </li>
-              <li>
-                <var>configuration</var>'s {{VideoConfiguration/framerate}}
-                contains one occurrence of U+002F SLASH character (/) and the
-                substrings before and after this character, when applying the
-                <a>rules for parsing floating-point number values</a> results in
-                a number that is finite and greater than 0.
-              </li>
-            </ul>
+            If {{VideoConfiguration/framerate}} is not finite or is not greater
```
Comment (chcunningham) [inline index.bs]: I take as implicitly banning NaN as well. Just calling it out in case that needed special mention.

**Comment by mwatson2** [comment]
Why is this proposed now ? I don't recall discussing changing this back.

File: index.bs [hunk 2]
```diff
@@ -372,7 +357,7 @@ spec: workers; urlPrefix: https://www.w3.org/TR/workers/#
         The <dfn for='VideoConfiguration' dict-member>framerate</dfn> member
         represents the framerate of the video track. The framerate is the number
         of frames used in one second (frames per second). It is represented
-        either as a double or as a fraction.
+        either as a double.
```
[the remark this answers is not in the document]
Comment (mounirlamouri) [reply index.bs]: Done.

File: index.bs (hunk 1 above)
Comment (mounirlamouri) [reply index.bs]: Yeah, it should.

**Comment by mounirlamouri** [comment]
> Why is this proposed now ? I don't recall discussing changing this back.

I believe this is the conclusion we reached in #96 

Is Netflix using the current string parsing behaviour or plan to use it?

**Comment by mwatson2** [comment]
@mounirlamouri In the thread on #96 it says that three people agreed in a f2f meeting (not sure when). I'm not sure how that counts as a conclusion.

**Comment by mounirlamouri** [comment]
> @mounirlamouri In the thread on #96 it says that three people agreed in a f2f meeting (not sure when). I'm not sure how that counts as a conclusion.

The fact that no one objected in months made me think that it was an accepted decision. Could you raise your concerns over there?

**Comment by chcunningham** [comment]
Discussed at TPAC, resolved to proceed with the revert to a double. https://www.w3.org/2019/09/18-mediawg-minutes.html

Mark felt ratio representation was less critical upon noting that the framerate is not intended to be used determine supported=true/false. Its main purpose is to help make smooth=true/false predictions.

**Pull request merged**